Linux/sgx/fips_test/sgx_t.mk

```diff
@@ -0,0 +1,187 @@
+Enclave_Cpp_Flags :=  $(Common_C_Cpp_Flags) -std=c++11 -nostdinc++
+
+SgxSSL_Link_Libraries := -L$(OPENSSL_LIBRARY_PATH) -Wl,--whole-archive -l$(SGXSSL_Library_Name) -Wl,--no-whole-archive \
+						 -l$(OpenSSL_Crypto_Library_Name)
+Security_Link_Flags := -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -pie
+
+Enclave_Link_Flags := $(SGX_COMMON_CFLAGS) -Wl,--no-undefined -nostdlib -nodefaultlibs -nostartfiles \
+    $(Security_Link_Flags) \
+    $(SgxSSL_Link_Libraries) -L$(SGX_LIBRARY_PATH) \
+    -Wl,--whole-archive -l$(Trts_Library_Name) -l$(SGXSSL_FIPS_TLIB) -Wl,--no-whole-archive \
+    -Wl,--start-group -lsgx_tstdc -lsgx_pthread -lsgx_tcxx -lsgx_tcrypto $(TSETJMP_LIB) -l$(Service_Library_Name) -Wl,--end-group \
+    -Wl,-Bstatic -Wl,-Bsymbolic -Wl,--no-undefined \
+    -Wl,-pie,-eenclave_entry -Wl,--export-dynamic  \
+    -Wl,--defsym,__ImageBase=0 \
+    -Wl,--version-script=$(ENCLAVE_DIR)/enclave.lds
+
+Enclave_Test_Key := $(ENCLAVE_DIR)/enclave_private.pem
+
+# OpenSSL configuration file
+OPENSSLCONF:=openssl.cnf
+FIPSMODULECONF:=fipsmodule.cnf
+LIBDIR := lib64
+
+.PHONY: all clean install_conf
+
+all: enclave.signed.so
+
+######## Enclave Objects ########
+SGXSSL_ADDTIONAL_EDL_PATH=$(PACKAGE_INC)/filefunc
+
+$(ENCLAVE_DIR)/enclave_t.c: $(SGX_EDGER8R) $(ENCLAVE_DIR)/enclave.edl
+	@cd $(ENCLAVE_DIR) && $(SGX_EDGER8R) --trusted enclave.edl --search-path $(PACKAGE_INC) --search-path $(SGX_SDK_INC) --search-path $(SGXSSL_ADDTIONAL_EDL_PATH)
+	@echo "GEN  =>  $@"
+
+$(ENCLAVE_DIR)/enclave_t.o: $(ENCLAVE_DIR)/enclave_t.c
+	$(VCC) $(Enclave_C_Flags) -c $< -o $@
+	@echo "CC   <=  $<"
+
+$(ENCLAVE_DIR)/%.o: $(ENCLAVE_DIR)/%.cpp $(ENCLAVE_DIR)/enclave_t.c
+	$(VCXX) $(Enclave_Cpp_Flags) -c $< -o $@
+	@echo "CXX  <=  $<"
+
+$(ENCLAVE_DIR)/%.o: $(ENCLAVE_DIR)/%.c $(ENCLAVE_DIR)/enclave_t.c
+	$(VCC) $(Enclave_C_Flags) -c $< -o $@
+	@echo "CC  <=  $<"
+
+$(ENCLAVE_DIR)/tests/%.o: $(ENCLAVE_DIR)/tests/%.c $(ENCLAVE_DIR)/enclave_t.c
+	$(VCC) $(Enclave_C_Flags) -c $< -o $@
+	@echo "CC  <=  $<"
+
+enclave.so: $(ENCLAVE_DIR)/enclave_t.o $(Enclave_Cpp_Objects) $(Enclave_C_Objects)
+	$(VCXX) $^ -o $@ $(Enclave_Link_Flags)
+	@echo "LINK =>  $@"
+
+enclave.signed.so: enclave.so install_conf
+ifeq ($(wildcard $(Enclave_Test_Key)),)
+	@echo "There is no enclave test key <enclave_private.pem>."
+	@echo "The project will generate a key <enclave_private.pem> for testing."
+	@openssl genrsa -out $(Enclave_Test_Key) -3 3072
+endif
+	@echo "SIGN =>  $@"
+	$(SGX_ENCLAVE_SIGNER) sign -key $(Enclave_Test_Key) -enclave enclave.so -out $@ -config $(ENCLAVE_DIR)/enclave.config.xml
+
+install_conf:
+	@echo "*** Installing OpenSSL configuration"
+	@echo "install $(OPENSSLCONF) -> $(SGX_SDK)/$(LIBDIR)/$(OPENSSLCONF)"
+	@cp -f $(OPENSSLCONF).tmpl $(OPENSSLCONF)
+	echo ".include $(SGX_SDK)/$(LIBDIR)/$(FIPSMODULECONF)" >> $(OPENSSLCONF)
+
+clean:
+	@rm -f enclave.* $(ENCLAVE_DIR)/enclave_t.* $(Enclave_Cpp_Objects) $(Enclave_C_Objects) $(Enclave_Test_Key)
+	@rm -f fips.so openssl.cnf
```

Comment:
No fips.so is copied, right?

Reply:
Not in this makefile.
The fips.so is copied in the open_source/Makefile lines 73--77.